internal/codecs/codecs.go

```diff
@@ -102,6 +102,8 @@ func BuildEncoder(
 		return buildArrayEncoder(desc, version)
 	case descriptor.Range:
 		return buildRangeEncoder(desc, version)
+	case descriptor.MultiRange:
+		return buildMultiRangeEncoder(desc, version)
```

Comment:
looks like the same thing needs to be done in `BuildEncoderV2` below and also you need to add lines to build decoders in `BuildDecoder` and `BuildDecoderV2` in the same file